sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -620,6 +622,12 @@ class SparkSession private(
    */
   @transient lazy val catalog: Catalog = new CatalogImpl(self)
 
+  @transient private lazy val catalogs = new mutable.HashMap[String, CatalogPlugin]()
```

Comment:
Should there be some best-effort lifecycle for stopping or closing these catalog plugins? I'm wondering for example if we want to close connections to databases when the Spark Session shuts down. Maybe add a `stop()` API and calling it from `SparkSession#stop`?

Reply:
We can check for `Closeable` and clean them up. Do you think this is needed?

Reply:
Using an explicit `stop` API is more consistent with other parts of Spark, thinking the SparkSession itself and scheduler components for example.

Reply:
Is there a use case for stopping catalogs? HiveExternalCatalog doesn't have a stop method.

Reply:
possibly - a job server type thing that would like to "update" or "refresh" catalog perhaps. I don't know if it's super important for starter

Reply:
I would say let's build it when we have a use case for it and avoid unnecessary additions.

Reply:
I was thinking JDBC would be a significant one, enough so that it's worth putting in the API up front and showing that we support it. We'd like to be able to shut down connection pools when the catalog is discarded.